### PR TITLE
replace pkg_resources.load_entry_point

### PR DESCRIPTION
**Pull request by gotmax23**
pkg_resources is deprecated. Use the corresponding importlib.metadata interface instead. Use the stdlib version on python >= 3.8 and use the importlib_metadata backport on older versions.

Relates: https://github.com/benoitc/gunicorn/issues/2747
Closes: https://github.com/benoitc/gunicorn/pull/2958

### CODE REVIEW

**Comment by Htermotto** [comment]
Any interest in getting this merged?

**Comment by benoitc** [comment]
this need to be tested.

**Comment by Htermotto** [comment]
Got it, what's the best way to go about that? Does Gunicorn have an automated build/test suite?

File: gunicorn/util.py [hunk 1]
```diff
@@ -53,6 +56,14 @@ def _setproctitle(title):
     def _setproctitle(title):
         pass
 
+def load_entry_point(distribution, group, name):
```
Comment (mattwang44) [inline gunicorn/util.py:59]: `gunicorn\util.py:59:1: E302 expected 2 blank lines, found 1`
```suggestion

def load_entry_point(distribution, group, name):
```

File: gunicorn/util.py [hunk 2]
```diff
@@ -22,7 +22,10 @@
 import traceback
 import warnings
 
-import pkg_resources
+try:
+    import importlib.metadata as importlib_metadata
+except:
```
Comment (mattwang44) [inline gunicorn/util.py:27]: `gunicorn\util.py:27:1: E722 do not use bare 'except'`
```suggestion
except Exception:
```

**Review by mattwang44 — COMMENTED** [review]
here are the errors found by pycodestyle

**Comment by benoitc** [comment]
testing it asap. What's your thoughts with #2958 ?

**Comment by gotmax23** [comment]
> thanks testing it asap. What's your thoughts with #2958 ?

#2958 looks good to me. I think there's a conflict with this one, though. @taleinat, do you mind if I cherry pick the commits from your PR into this one?

**Comment by taleinat** [comment]
Instead of cherry-picking, I think it would be better to merge my branch into this PR's branch.

**Review by taleinat — APPROVED** [review]
LGTM.

Note that this requires the changes from PR #2963 too, since the changes there are also needed in order to completely remove the dependency on setuptools.

**Comment by gotmax23** [comment]
> I don't see real conflicts, only git not understanding how to combine the changes to `setup.py`.

Yes, that's a conflict.

> Instead of cherry-picking, I think it would be better to merge my branch into this PR's branch.

I checked out your branch and applied my commit on top of yours to avoid merge commits and preserve your PGP signature.

**Comment by benoitc** [comment]
thanks @gotmax23 @taleinat for your patches :)

**Comment by gotmax23** [comment]
Thanks @taleinat and @benoitc!

**Pull request merged**